### PR TITLE
cache: fix null pointer access when cachemanager.Disable is true

### PR DESCRIPTION
**Pull request by jiangliu**
Filesystem.cacheMgr is null when cacheManager.Disable is configured to be true, thus cause null pointer access. Fix it by always create the cacheMgr object no matter cacheManager.Disable is true or false.

### CODE REVIEW

**Comment by sctb512** [comment]
Why not check if the cacheMgr is nil?
It seems `WithEnableStargz` and `referrerMgr`,  also have the same problem.

File: pkg/cache/manager.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type Manager struct {
 }
 
 type Opt struct {
+	Disabled bool
```
Comment (imeoer) [inline pkg/cache/manager.go:40]: Good catch! Maybe also update these `config.toml`:

- misc/snapshotter/config.toml
- tests/e2e/k8s/snapshotter-cri.yaml
- tests/e2e/k8s/snapshotter-kubeconf.yaml

**Comment by jiangliu** [comment]
> Why not check if the cacheMgr is nil? It seems `stargzResolver` and `referrerMgr` also have the same problem.

With the evolution of nydus-snapshotter, `cache manager` becomes mandatory instead optional now. On the other hand, `stargzResolver` and `referrerMgr` are optional.

File: pkg/cache/manager.go (hunk 1 above)
Comment (jiangliu) [reply pkg/cache/manager.go:40]: `cache_manager.disable` is still set to be `false` for those three files. I'm keeping the configuration item for compatibility though it doesn't take effect. Not sure whether `cache_manager.disable` will be used to control `auto garbage collect` in future or not.

Comment (jiangliu) [reply pkg/cache/manager.go:40]: After a second thought, I think we can support `recyclebin` to keep deleted files for a while. Those configuration item can be used for it.

**Review by sctb512 — APPROVED** [review]
LGTM

File: pkg/cache/manager.go (hunk 1 above)
Comment (imeoer) [reply pkg/cache/manager.go:40]: Introducing recyclebin seems to add complexity and doesn't feel necessary, it is better to let kubelet & containerd control the GC policy.

**Review by changweige — APPROVED** [review]
LGTM

**Review by changweige — APPROVED** [review]
LGTM

**Pull request merged**